src/workerd/io/io-context.c++

```diff
@@ -193,13 +201,18 @@ IoContext::IoContext(ThreadContext& thread,
 
     // Stop the ActorCache from flushing any scheduled write operations to prevent any unnecessary
     // or unintentional async work
-    localAbortPromise = localAbortPromise.catch_([this](kj::Exception&& e) -> kj::Promise<void> {
-      KJ_IF_MAYBE(a, actor) {
-        a->shutdownActorCache(e);
+    localAbortPromise = (kj::coCapture([this, promise = kj::mv(localAbortPromise)]() mutable
+        -> kj::Promise<void> {
+      try {
+        co_await promise;
+      } catch (...) {
+        auto exception = kj::getCaughtExceptionAsKj();
+        KJ_IF_MAYBE(a, actor) {
+          a->shutdownActorCache(exception);
+        }
+        throw exception;
```

Comment:
Nit: `kj::throwFatalException()` -- as I recall, it's a required part of the `catch (...) { auto e = kj::getCaughtExceptionAsKj()` dance.